### PR TITLE
Update Bucket references to BucketV2

### PR DESCRIPTION
**Pull request by t0yv0**
### Proposed changes

Update references to aws.s3.Bucket with aws.s3.BucketV2, just as we would like our users to do.

#### Content with onObjectCreated and similar overlays

There is a bit of a migration gap for this content we need to think through: https://github.com/pulumi/docs/issues/13014

#### Policy (cross-guard)

Separating this out as it requires extra effort: https://github.com/pulumi/docs/issues/13013

#### Shortcodes and Layouts

I have found the following references to Bucket but was not sure how these features work to update them safely:

layouts/page/crossguard.html
layouts/partials/home/cli.html
layouts/partials/home/console.html
layouts/partials/home/ide.html
layouts/shortcodes/pulumi-bucket.html
layouts/shortcodes/pulumi-new-bucket.html

### Related issues (optional)

Related to https://github.com/pulumi/home/issues/3633

### CODE REVIEW

**Comment by t0yv0** [comment]
There is still a little bit of content here that's update-able but entangled with ACLs and websites, that doesn't work very well anymore due to AWS changes. I'm going to try to get it over the finish line tomorrow:

** TODO content/docs/iac/concepts/how-pulumi-works.md                  
** TODO content/docs/iac/get-started/aws/deploy-changes.md
** TODO content/docs/iac/get-started/aws/deploy-stack.md  
** TODO content/docs/iac/packages-and-automation/automation-api/getting-started-automation-api.md

**Comment by t0yv0** [comment]
https://github.com/pulumi/docs/pull/13142 takes care of how-pulumi-works page.

File: content/docs/iac/adopting-pulumi/import/_index.md [hunk 1]
```diff
@@ -77,13 +77,13 @@ For help identifying a resource's type token and lookup property, see [Where to
 In this example, a previously provisioned Amazon S3 bucket named `company-infra-logs` is imported into a Pulumi stack named `dev` (the currently [selected](/docs/cli/commands/pulumi_stack_select/) stack) and given a resource name of `infra-logs`:
 
 ```bash
-$ pulumi import aws:s3/bucket:Bucket infra-logs company-infra-logs
+$ pulumi import aws:s3/bucket:BucketV2 infra-logs company-infra-logs
```
Comment (sean1588) [inline content/docs/iac/adopting-pulumi/import/_index.md:80]:
```suggestion
$ pulumi import aws:s3/bucketv2:BucketV2 infra-logs company-infra-logs
```

should this be bucketv2 instead?

File: content/docs/iac/concepts/resources/components.md [hunk 2]
```diff
@@ -142,48 +142,48 @@ This example demonstrates both the naming convention and how to designate the co
...
+let bucket = new aws.s3.BucketV2(`${name}-bucket`,
     {/*...*/}, { parent: this });
 ```
 
 {{% /choosable %}}
 {{% choosable language python %}}
 
 ```python
-bucket = s3.Bucket(f"{name}-bucket",
+bucket = s3.BucketV2(f"{name}-bucket",
```
Comment (sean1588) [inline content/docs/iac/concepts/resources/components.md:161]: not sure why there is an "f" here.

File: content/docs/iac/get-started/aws/modify-program.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ In `main.go`, create the `BucketObject` right after creating the bucket itself:
 ```go
 // Create an S3 Bucket object
 _, err = s3.NewBucketObject(ctx, "index.html", &s3.BucketObjectArgs{
-    Bucket:  bucket.ID(),
+    Bucket:  bucket.Bucket(),
```
Comment (sean1588) [inline content/docs/iac/get-started/aws/modify-program.md:109]:
```suggestion
    Bucket:  bucket.ID(),
```

File: content/docs/iac/get-started/aws/modify-program.md [hunk 4]
```diff
@@ -124,7 +124,7 @@ In `Program.cs`, create a new `BucketObject` right after creating the bucket its
 // Create an S3 Bucket object
 var bucketObject = new BucketObject("index.html", new BucketObjectArgs
 {
-    Bucket = bucket.BucketName,
+    Bucket = bucket.Bucket,
```
Comment (sean1588) [inline content/docs/iac/get-started/aws/modify-program.md:127]:
```suggestion
    Bucket = bucket.Id,
```

File: content/docs/iac/adopting-pulumi/import/_index.md (hunk 1 above)
Comment (t0yv0) [reply content/docs/iac/adopting-pulumi/import/_index.md:80]: I think you are right, good catch!

File: content/docs/iac/concepts/resources/components.md (hunk 2 above)
Comment (t0yv0) [reply content/docs/iac/concepts/resources/components.md:161]: This is a Python feature called f-strings https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals - I think this is quite idiomatic these days.

File: content/docs/iac/get-started/aws/modify-program.md (hunk 3 above)
Comment (t0yv0) [reply content/docs/iac/get-started/aws/modify-program.md:109]: Thanks for spotting!

File: content/docs/iac/get-started/aws/modify-program.md (hunk 4 above)
Comment (t0yv0) [reply content/docs/iac/get-started/aws/modify-program.md:127]: Thank you!

**Comment by t0yv0** [comment]
Let me rebase and follow up on these, thanks so much!

File: content/docs/iac/concepts/resources/components.md (hunk 2 above)
Comment (sean1588) [reply content/docs/iac/concepts/resources/components.md:161]: ahh thank you!!

**Comment by t0yv0** [comment]
I had some difficulties with pulumi_org_search.md that Git considers to be binary; but I realized it has auto-generated content. We actually need to fix Pulumi CLI there.  I have opened https://github.com/pulumi/pulumi/pull/17692

**Comment by t0yv0** [comment]
Added the suggested edits, rebased on master and excluded conflicting  pulumi_org_search.md , PTAL.

**Comment by t0yv0** [comment]
Done additional manual checks on edited examples via `pulumi preview`. @sean1588  - had to back one out and added a few small fixes, PTAL.

```
** DONE review_project.md
** DONE languages-sdks/python
** DONE languages-sdks/yaml
** DONE packages-and-automation/continuous-delivery
** DONE pulumi-cloud/insights/export.md
** DONE pulumi-cloud/insights/search.md
** DONE pulumi-cloud/reference/cloud-rest-api
** DONE migrate/terraform.md
** DONE resources/how-to-deploy-a-django
** DONE tutorials/abstration-encapsulation/component-resources/index.md
** DONE tutorials/abstration-encapsulation/encapsulation/index.md
** DONE tutorials/abstration-encapsulation/stack-outputs-and-references/index.md
** DONE what-is/guide-to-automating-
** DONE what-is/what-is-yaml

```

**Review by sean1588 — APPROVED** [review]
thanks again @t0yv0 ❤️

**Pull request merged**